frame/dapps-staking/src/lib.rs

```diff
@@ -97,6 +130,128 @@ impl Default for Version {
     }
 }
 
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct EraStake<Balance: AtLeast32BitUnsigned + Copy> {
+    /// Staked amount in era
+    staked: Balance,
+    /// Staked era
+    era: EraIndex,
+}
+
+impl<Balance: AtLeast32BitUnsigned + Copy> EraStake<Balance> {
+    fn new(staked: Balance, era: EraIndex) -> Self {
+        Self { staked, era }
+    }
+}
+
+/// Contains information about how much was staked in each era
+#[derive(Encode, Decode, Clone, Default, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct StakerInfo<Balance: AtLeast32BitUnsigned + Copy> {
+    // Size of this list would be limited by a configurable constant TODO
+    stakes: Vec<EraStake<Balance>>,
+}
+
+impl<Balance: AtLeast32BitUnsigned + Copy> StakerInfo<Balance> {
+    /// true if no stakes exist, false otherwise
+    fn is_empty(&self) -> bool {
+        self.stakes.is_empty()
+    }
+
+    /// number of `EraStake` chunks
+    fn len(&self) -> u32 {
+        self.stakes.len() as u32
+    }
+
+    /// Stakes some value in the specified era.
+    /// User should ensure that given era is either equal or greater than the
+    /// latest available era in the staking info.
+    fn stake(&mut self, current_era: EraIndex, value: Balance) -> Result<(), &str> {
+        if self.stakes.is_empty() {
+            self.stakes.push(EraStake::new(value, current_era))
+        } else {
+            let era_stake = self.stakes.last().unwrap(); // exists if vec not empty
+            if era_stake.era > current_era {
```

Comment:
When can this happen?

Reply:
Initially, I implemented `stake` and `unstake` without this check.
But since we have to be 100% sure nothing can get messed up, I added this check.

To answer your question - it cannot happen :).
It's just a safety measure in case I have a bug in the code that would allow it.